lib/discourse_theme/cli_commands/rspec.rb

```diff
@@ -0,0 +1,242 @@
+# frozen_string_literal: true
+
+require "selenium-webdriver"
+
+module DiscourseTheme
+  module CliCommands
+    class Rspec
+      DISCOURSE_TEST_DOCKER_CONTAINER_NAME_PREFIX = "discourse_theme_test"
+      DISCOURSE_THEME_TEST_TMP_DIR = "/tmp/.discourse_theme_test"
+      SELENIUM_HEADLESS_ENV = "SELENIUM_HEADLESS=0"
+
+      class << self
+        def discourse_test_docker_container_name
+          "#{DISCOURSE_TEST_DOCKER_CONTAINER_NAME_PREFIX}_#{DiscourseTheme::VERSION}"
+        end
+
+        def run(settings:, dir:, args:, reset: false)
+          settings.local_discourse_directory = nil if reset
+
+          spec_path = "/spec"
+          index = dir.index(spec_path)
+
+          if index
+            spec_path = dir[index..-1]
+            dir = dir[0..index - 1]
+          end
+
+          spec_directory = File.join(dir, "/spec")
+
+          unless Dir.exist?(spec_directory)
+            raise DiscourseTheme::ThemeError.new "'#{spec_directory} does not exist"
+          end
+
+          configure_local_directory(settings)
+
+          headless = !!args.delete("--headless")
+
+          if settings.local_discourse_directory.empty?
+            run_tests_with_docker(
+              Pathname.new(dir).basename.to_s,
+              spec_directory,
+              spec_path,
+              headless: headless,
+              verbose: !!args.delete("--verbose"),
+              rebuild: !!args.delete("--rebuild"),
+            )
+          else
+            run_tests_locally(
+              settings.local_discourse_directory,
+              spec_directory,
+              headless: headless,
+            )
+          end
```

Comment:
This portion needs to be reviewed as we added a new conditional based on the settings selected by the user.